waiting_room.go

```diff
@@ -209,6 +220,29 @@ func (api *API) WaitingRoomStatus(ctx context.Context, zoneID, waitingRoomID str
 	return r.Result, nil
 }
 
+// WaitingRoomPagePreview uploads a custom waiting room page for preview and
+// returns a preview URL.
+//
+// API reference: https://api.cloudflare.com/#waiting-room-create-a-custom-waiting-room-page-preview
+func (api *API) WaitingRoomPagePreview(ctx context.Context, zoneID, customHTML string) (WaitingRoomPagePreview, error) {
+	uri := fmt.Sprintf("/zones/%s/waiting_rooms/preview", zoneID)
+	res, err := api.makeRequestContext(ctx, http.MethodPost, uri, struct {
+		CustomHTML string `json:"custom_html"`
```

Comment:
we should not be using an anonymous struct here; you should extract this to a named struct for reuse.

Reply:
I created two structs `WaitingRoomPagePreviewURL` and `WaitingRoomPagePreviewCustomHTML`. Not sure if there would be a better naming scheme.